### PR TITLE
Adds prohibited attributes expectation to rule 5c01ea

### LINKED ISSUE #1535: Consider prohibited ARIA attributes in ARIA props permitted [5c01ea]
WAI-ARIA 1.2 comes with [Prohibited States and Properties](https://www.w3.org/TR/wai-aria-1.2/#prohibitedattributes). When ARIA 1.2 is published as candidate rec, I think we should add this as an expectation to the [ARIA states and properties permitted](https://act-rules.github.io/rules/5c01ea) rule.

### PR DESCRIPTION
**Pull request by schne324**
This patch adds a second expectation regarding usage of prohibited attributes for a given role. Based on [ARIA 1.2's statement](https://www.w3.org/TR/wai-aria-1.2/#state_property_processing):

> Global states and properties are supported on any element in the host language. However, authors MUST only use non-global states and properties on elements with a role supporting the state or property; either defined as an explicit WAI-ARIA role, or as defined by the host language implicit WAI-ARIA semantic matching an appropriate WAI-ARIA role.

Closes issue(s):

- closes #1535

### CODE REVIEW

File: _rules/aria-state-or-property-permitted-5c01ea.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ For each test target, one of the following is true:
 - **semantic Role**: the test target is an [inherited][], [supported][], or [required][] [state][] or [property][] of the [semantic role][] of the element on which the test target is specified; or
 - **language feature**: the test target is specified on an [HTML element][namespaced element] and is allowed on that element. Which ARIA states or properties may be used on which element is described in [ARIA in HTML](https://w3c.github.io/html-aria/).
 
+For each test target there are no prohibited attributes for the target's role.
```
Comment (Jym77) [inline _rules/aria-state-or-property-permitted-5c01ea.md:45]: 1. In this rule, the "test targets" are the ARIA states and properties, not the element. So they do not have a role. This should be phrased as "the test target is not prohibited on the role of the element on which it is specified" or something like that (see the way the second bullet is phrased).
2. This sounds like a completely different expectation, should they be numbered as "Expectation 1", "Expectation 2" instead?
3. What is a "prohibited attribute"? This needs to link to some specification. "role" should also link to our definition of "semantic role".

Comment (Jym77) [inline _rules/aria-state-or-property-permitted-5c01ea.md:45]: …and I actually do not understand the need for this new expectation 🤔 

If the state or property is either global, inherited, supported, required, or explicitly allowed by ARIA in HTML, how can it be prohibited?
This needs at least a new test case to show something that fails because of this new expectation and wouldn't fail without it (see [Test every condition](https://act-rules.github.io/pages/design/test-cases/#test-every-condition) in our test cases design document).

**Review by Jym77 — CHANGES_REQUESTED** [review]

File: _rules/aria-state-or-property-permitted-5c01ea.md (hunk 1 above)
Comment (WilcoFiers) [reply _rules/aria-state-or-property-permitted-5c01ea.md:45]: It does need new tests. As for why this is needed; ARIA 1.2 prohibits aria-label(ledby) on a bunch of things, even though it is a global attribute. And aria-roledescription on `generic`.

File: _rules/aria-state-or-property-permitted-5c01ea.md [hunk 2]
```diff
@@ -34,14 +34,18 @@ acknowledgments:
...
 
 For each test target, one of the following is true:
 
 - **global**: the test target is a [global state or property][global]; or
 - **semantic Role**: the test target is an [inherited][], [supported][], or [required][] [state][] or [property][] of the [semantic role][] of the element on which the test target is specified; or
 - **language feature**: the test target is specified on an [HTML element][namespaced element] and is allowed on that element. Which ARIA states or properties may be used on which element is described in [ARIA in HTML](https://w3c.github.io/html-aria/).
 
+## Expectation 2
+
+Each test target is not [prohibited](https://www.w3.org/TR/wai-aria-1.2/#state_property_processing) on the role of the element on which it is specified.
```
Comment (schne324) [inline _rules/aria-state-or-property-permitted-5c01ea.md]:
```suggestion
Each test target is not [prohibited](https://www.w3.org/TR/wai-aria-1.2/#state_property_processing) on the [semantic role][] of the element on which it is specified.
```

File: _rules/aria-state-or-property-permitted-5c01ea.md (hunk 1 above)
Comment (carlosapaduarte) [reply _rules/aria-state-or-property-permitted-5c01ea.md]: I'm not sure I agree that it does not need new tests. Why not have a failed example with an `aria-label` on `generic` container? It passes expectation one (because it is a global property) and fails expectation 2 (because it is prohibited).

**Review by carlosapaduarte — DISMISSED** [review]
See https://github.com/act-rules/act-rules.github.io/pull/1877#discussion_r932079032

File: _rules/aria-state-or-property-permitted-5c01ea.md (hunk 1 above)
Comment (Jym77) [reply _rules/aria-state-or-property-permitted-5c01ea.md]: @carlosapaduarte "It does need new tests" and that is also what @WilcoFiers wrote 😃 
(and I also had to read it twice to see there is no negation in the first sentence, I have no clue why this sentence is apparently so hard to parse 🤔)

Comment (carlosapaduarte) [reply _rules/aria-state-or-property-permitted-5c01ea.md]: Damn... I read it more than twice and it always registered in my brain as "It does not need new tests" 🫣

File: _rules/aria-state-or-property-permitted-5c01ea.md [hunk 3]
```diff
@@ -34,14 +34,18 @@ acknowledgments:
...
 For each test target, one of the following is true:
 
 - **global**: the test target is a [global state or property][global]; or
 - **semantic Role**: the test target is an [inherited][], [supported][], or [required][] [state][] or [property][] of the [semantic role][] of the element on which the test target is specified; or
 - **language feature**: the test target is specified on an [HTML element][namespaced element] and is allowed on that element. Which ARIA states or properties may be used on which element is described in [ARIA in HTML](https://w3c.github.io/html-aria/).
 
+## Expectation 2
+
+Each test target is not [prohibited](https://www.w3.org/TR/wai-aria-1.2/#state_property_processing) on the [semantic role][] of the element on which it is specified.
+
```
Comment (Jym77) [inline _rules/aria-state-or-property-permitted-5c01ea.md]: I'm not fully sure we need to have it as a separate expectation. The [ARIA 1.2 definition of global](https://www.w3.org/TR/wai-aria-1.2/#global_states) states:
> The following global states and properties are supported by all roles and by all base markup elements **unless otherwise prohibited**.
(emphasis mine)
and later lists:
> * [aria-label](https://www.w3.org/TR/wai-aria-1.2/#aria-label) (Except where prohibited)
> * [aria-labelledby](https://www.w3.org/TR/wai-aria-1.2/#aria-labelledby) (Except where prohibited)

So, I do understand that prohibited attributes are not "global" for the roles on which they are prohibited (see also my comment at https://github.com/act-rules/act-rules.github.io/issues/1853#issuecomment-1195326291).

---
At the same time, I do agree that it is not completely obvious to derive that fact. This would at least require a note to explain that prohibited attributes are not global…

OTOH, I feel the current formulation also requires a note since it has a redundant expectation.

I think the less confusing thing to do would be to keep Expectation 2 as is, but add a Background note in the line of "prohibited attributes are not global, but we kept Expectation 2 for the sake of clarity"

File: _rules/aria-state-or-property-permitted-5c01ea.md [hunk 4]
```diff
@@ -181,6 +185,14 @@ The `aria-orientation` property may not be used on `audio` element, nor it can b
 <audio src="/test-assets/moon-audio/moon-speech.mp3" controls aria-orientation="horizontal"></audio>
 ```
 
+#### Failed Example 3
+
+The `aria-label` property may not be used on an element with a `generic` role as it is [prohibited from being named](https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited).
```
Comment (Jym77) [inline _rules/aria-state-or-property-permitted-5c01ea.md]: Since the expectation is in term of "prohibited" attributes rather than roles "prohibited from being named", I'd rather have the example description using the same language too.

**Review by Jym77 — DISMISSED** [review]
I think we should add an explanation on the link between global and prohibited.

File: _rules/aria-state-or-property-permitted-5c01ea.md (hunk 3 above)
Comment (WilcoFiers) [reply _rules/aria-state-or-property-permitted-5c01ea.md]: While yes, prohibited attributes might not be global... depending on how you read that, I think making this explicit is important for the clarity of the rule.

**Review by Jym77 — APPROVED** [review]

**Review by lbrnpm — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

**Comment by carlosapaduarte** [comment]
Merging
